### PR TITLE
Accumulation of changes to RRTMGP

### PR DESCRIPTION
**Pull request by dustinswales**
## Description
This PR contains improvements and bug fixes to the RRTMGP enabled ccpp-physics.

- Bug fix (indexing error) for temperature being passed to GP. (Only exercised in rare cases in EMC's C768L127 configuration).
- New coupling of RRTMGP to the RRFS physics. Based on fv3_regional_quilt using FV3_GFS_v15_thompson_mynn_RRTMGP.
- Revised approach to the LW flux adjustment using the full vertical profile of the Jacobian of the upwelling LW. Previously only the surface flux was adjusted.
- Move allocation of GP interstitial DDTs from the LW scheme to GFS_typedefs.F90 interstitial_create/reset.

### Issue(s) addressed
N/A

## Testing
This was tested on Hera using the Intel compilers.
There are no changes to the baselines for the current RTs; However, there are proposed changes to the default namelist setting in the associated UFS PR.
Not all changes are covered by the RT's. A new RT is proposed in the associated UFS PR that exercises the regional (RRFS) coupled to RRTMGP. *This will be the first RT that has RRTMGP coupled to the Thompson MP scheme*. New baseline needed.

## Dependencies
https://github.com/NCAR/ccpp-physics/pull/614

### CODE REVIEW

File: ccpp/CMakeLists.txt
Comment (climbfuji) [inline ccpp/CMakeLists.txt:181]: This *should* be ok because `ccpp/{framework,physics,data,driver}` each have their own logic. Have you checked that the computational performance is not affected by this change?

File: ccpp/CMakeLists.txt
Comment (climbfuji) [inline ccpp/CMakeLists.txt:195]: Thanks for removing this, not needed.

**Review by climbfuji — COMMENTED** [review]
This PR looks good to me, will approve when it is time to merge.

File: ccpp/suites/suite_FV3_GFS_v15_thompson_mynn_RRTMGP.xml
Comment (junwang-noaa) [inline ccpp/suites/suite_FV3_GFS_v15_thompson_mynn_RRTMGP.xml:3]: Do you also want to add a suite file for suite_FV3_GFS_v15_thompson_mynn_lam3km.xml? The suite_FV3_GFS_v15_thompson_mynn.xml is used for low resolution test (in RT), if you want to test it in high resolution LAM, you may need to add a new suite file for suite_FV3_GFS_v15_thompson_mynn_lam3km.xml

Comment (dustinswales) [reply ccpp/suites/suite_FV3_GFS_v15_thompson_mynn_RRTMGP.xml:3]: We don't need to do this at this time.

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by junwang-noaa — APPROVED** [review]
After the gitmodule is updated, the code can be merged.

**Review by climbfuji — APPROVED** [review]
Submodule pointer is correct, please merge.

**Comment by climbfuji** [comment]
We'll have to wait for the fv3atm submodule pointer update after the fv3atm merge

**Pull request merged**
